### PR TITLE
"from source" third-party dependency management via cmake

### PR DESCRIPTION
**Pull request by marpaia**

### CODE REVIEW

**Comment by marpaia** [comment]
JFYI, this PR is going to fail build for a while.

File: third-party2/python.cmake [hunk 1]
```diff
@@ -0,0 +1,9 @@
+ExternalProject_Add(
+  python
+  URL https://www.python.org/ftp/python/2.7.9/Python-2.7.9.tgz
+  URL_HASH SHA1=7a191bcccb598ccbf2fa6a0edce24a97df3fc0ad
+  BUILD_IN_SOURCE 1
+  CONFIGURE_COMMAND ./configure --prefix=${third_party_prefix}
+  BUILD_COMMAND make
+  INSTALL_COMMAND make install
```
Comment (marpaia) [inline third-party2/python.cmake]: i think "make" and "make install" are the defaults here so these lines can probably be omitted for the sake of brevity.

**Comment by theopolis** [comment]
If you build in source for some packages do you need to make clean if you update the URL hash?

**Comment by marpaia** [comment]
I don't believe so, it should notice that the cmake file changed and rebuild it (as well as everything that explicitly depends on it)

**Pull request closed without merging**